src/uu/stty/src/stty.rs

```diff
@@ -462,6 +496,68 @@ fn apply_baud_rate_flag(termios: &mut Termios, input: &str) -> ControlFlow<bool>
     ControlFlow::Continue(())
 }
 
+fn apply_char_mapping(
+    termios: &mut Termios,
+    control_char_index: SpecialCharacterIndices,
+    new_val: &str,
+) -> Result<(), ControlCharMappingError> {
+    match string_to_control_char(new_val) {
+        Ok(val) => {
+            termios.control_chars[control_char_index as usize] = val;
+            Ok(())
+        }
+        Err(e) => Err(e),
+    }
```

Comment:
I think you can simplify it to something like:
```suggestion
    string_to_control_char(new_val).and_then(|val| {
        termios.control_chars[control_char_index as usize] = val;
        Ok(())
    })
```